### DISCUSSION TITLE
rapidly slowing clusters

### DISCUSSION THREAD

**Discussion by caffeinism** [rook/rook#10198] [Q&A] [answered]
Hello. I'm new to ceph, practicing a small storage cluster.
I've been experiencing clusters that are initially fast and then suddenly slow down after a few moments. Please refer to the Rados bench results below

```
$ rados bench -p blockpool 30 write -b 4096 -t 1024 
sec Cur ops   started  finished  avg MB/s  cur MB/s last lat(s)  avg lat(s)
    0       0         0         0         0         0           -           0
    1    1024    113866    112842   440.766   440.789  0.00711655  0.00895262
    2    1023    231677    230654   450.467   460.203   0.0108136   0.0088197
    3    1024    346470    345446   449.768   448.406  0.00613637  0.00885223
    4    1023    460057    459034   448.244   443.703   0.0198604  0.00888447
    5    1024    573418    572394   447.151   442.812  0.00705998  0.00892129
    6    1023    687437    686414   446.852   445.391  0.00733748  0.00893032
    7    1024    799072    798048   445.307    436.07  0.00934323  0.00896444
    8    1024    911428    910404     444.5   438.891  0.00753398  0.00898298
...
  166    1023  15578951  15577928   366.543   320.789   0.0129585   0.0109112
  167    1024  15661579  15660555    366.28   322.762   0.0077492   0.0109191
  168    1024  15758742  15757718   366.359   379.543  0.00835554   0.0109167
  169    1023  15838422  15837399   366.033   311.254   0.0415203   0.0109202
  170    1023  15873708  15872685    364.69   137.836   0.0172864   0.0109621
  171    1024  15906272  15905248   363.301   127.199  0.00424219   0.0109987
  172    1024  15941651  15940627   361.992   138.199  0.00603028   0.0110456
  173    1024  15976408  15975384   360.685    135.77  0.00645702   0.0110843
  174    1024  15992959  15991935   358.983   64.6523    0.346822   0.0111262
  175    1024  16006608  16005584   357.236   53.3164  0.00722037   0.0111888
  176    1024  16022312  16021288   355.555   61.3438  0.00666629   0.0112394
  177    1023  16039171  16038148   353.918   65.8594   0.0199628   0.0112978
  178    1024  16054709  16053685   352.271   60.6914  0.00323296   0.0113354
  179    1024  16071025  16070001   350.659   63.7344  0.00599297   0.0113929
2022-04-28T04:54:55.115916+0000 min lat: 0.00251709 max lat: 0.830688 avg lat: 0.0114354
  sec Cur ops   started  finished  avg MB/s  cur MB/s last lat(s)  avg lat(s)
  180    1024  16093716  16092692   349.203   88.6367  0.00755006   0.0114354
  181    1024  16115465  16114441   347.743    84.957  0.00792256   0.0114961
  182    1024  16130493  16129469   346.154   58.7031  0.00510463   0.0115456
  183    1024  16142851  16141827   344.526   48.2734    0.004796   0.0115973
  184    1023  16154228  16153205   342.895   44.4453  0.00581404   0.0116513
  185    1023  16169521  16168498   341.365   59.7383  0.00918128   0.0117089
  186    1024  16184937  16183913   339.853   60.2148  0.00682822   0.0117554
```
In the beginning, the performance is 3~400MB/s. However, at a certain point, it sharply decreases, resulting in an average performance of 60 MB/s.

I looked at various logs of osd to solve this. Bluestore's logs don't seem to have any problems. Among the logs of osd, there are the following parts.
```
        "op_w_latency": {
            "avgcount": 14863423,
            "sum": 372381.777428963,
            "avgtime": 0.025053567
        },
        "op_w_process_latency": {
            "avgcount": 14863423,
            "sum": 365172.947647443,
            "avgtime": 0.024568563
        },
        "op_w_prepare_latency": {
            "avgcount": 14863468,
            "sum": 2943.291376725,
            "avgtime": 0.000198021
        },
```

And note the following comments:
```
ceph_osd_op_w_latency           Latency of write operation (including queue time)
ceph_osd_op_w_process_latency   Latency of write operation (excluding queue time)
ceph_osd_op_prepare_latency     Latency of client operations (excluding queue time and wait for finished)
```

queue time + actual write time + "wait for finished" = 25ms
actual write time + "wait for finished" = 24.5ms
actual write time = 0.2ms

So I tried to ask what "wait for finished" is, but it seems like the time when it's non-blocking

I'm falling more and more into a labyrinth

Too difficult for me ceph!

**Comment by travisn** [comment, marked as answer]
This would be better opened as a ceph tracker, but perhaps @markhpc has some insight.

**Reply by markhpc** [reply]
There's a couple of potential causes.  One is that rocksdb can write metadata quickly to the WAL, but if flushes to the DB are slow (say due to backed up compactions), eventually it may start throttling writes so you don't run out of WAL space and completely stall out.  You may want to look at the OSD logs and see if there are any warnings about rocksdb throttling WAL writes.  You can also look at the occurrence of rocksdb compactions with this tool:

https://github.com/ceph/cbt/blob/master/tools/ceph_rocksdb_log_parser.py

You can download it and give it an OSD log as the parameter.

If that's not the issue, you might have to start trying to uncover why the latency is that high.  Is the underlying device saturated?  Is the network saturated?  Are write ops waiting on something? etc.

Good luck!
Mark

**Reply by caffeinism** [reply]
The ceph community feels like a huge barrier to entry. I was wondering if I could get some help from the rook community.

Thanks for your help. @markhpc It's maxing out the CPU in the initial setup. But after that, it slows down and doesn't fully utilize the CPU. Network bandwidth is sufficient and disk usage is low. So I fell into a labyrinth. I'll try the method you told me. Thank you again.

**Reply by caffeinism** [reply]
Hello again! I raised the rocksdb log level in osd and parsed the log with a little bit of code you suggested. The contents are as follows.
```
start   compaction output num    total     num     num     output (MB/s)      input (r/s)        output (r/s)       output/input ratio
offset  time       level  output output    input   output                                                                                                                       
        seconds           files  size      records records                                                                                                                         
198.308 0.577623   1      2      91008535  1576263 406538  150.2580444394688  2728878.524573987  703812.0019459059  0.25791254378235107
226.048 0.919202   1      2      90691697  2465910 406991  94.09285678251109  2682663.875840131  442765.5727467956  0.16504698062784123
239.001 0.001722   1      1      5697      328     148     3.155100221002559  190476.19047619047 85946.5737514518   0.45121951219512196
241.209 1.593177   1      5      300523841 563467  562981  179.8933003862443  353675.0781614347  353370.027310211   0.9991374827629657
242.537 1.321751   1      5      300852563 564047  563607  217.0721734714733  426742.2532685809  426409.3615211943  0.9992199231624315
243.885 1.343868   1      5      300409897 563216  562766  213.18552355078546 419100.685483991   418765.8311679421  0.999201017016562
254.529 0.915761   1      2      90563071  2446313 406519  94.31246234287732  2671344.3791556964 443913.85962057783 0.16617620067423916
282.657 0.994387   1      2      90400754  2443754 406391  86.6995216887988   2457548.218148467  408684.9486165849  0.1662978352158196
310.473 0.998574   1      2      90708566  2434462 406458  86.62996400090651  2437938.50030143   407038.43681089237 0.16696009220928484
315.973 2.879875   1      8      521696528 975226  975226  172.76047735124408 338634.83658144885 338634.83658144885 1.0
318.729 2.742516   1      8      521479348 974816  974816  181.33766976159006 355445.87524740055 355445.87524740055 1.0
321.361 2.62247    1      8      521833185 975492  975492  189.7672446140361  371974.5125778369  371974.5125778369  1.0
350.685 0.430079   1      2      90968130  2431164 407430  201.71635720996582 5652831.224030934  947337.5821651372  0.16758639071654566
467.195 0.528455   1      2      90764313  2417242 406661  163.7974740599633  4574168.09378282   769528.1528228516  0.16823346607414566
567.588 0.001354   1      1      5697      157     148     4.0126163815113784 115952.73264401773 109305.76070901034 0.9426751592356688
569.86  1.83404    1      8      537118924 1001696 1001696 279.2940844662933  546169.112996445   546169.112996445   1.0
571.496 1.626782   1      8      537173868 1001802 1001802 314.9093863567467  615818.2227243724  615818.2227243724  1.0
573.068 1.566346   1      8      537065722 1001606 1001606 326.9940264115202  639453.8626842345  639453.8626842345  1.0
573.508 0.42611    2      2      135710886 253654  253654  303.73374582839295 595278.2145455399  595278.2145455399  1.0
573.952 0.441106   2      2      135708683 253649  253648  293.40314003984867 575029.5847256668  575027.3176968801  0.9999960575440865
574.468 0.513229   2      2      135712315 253642  253641  252.1785581787753  494208.2384276804  494206.2899797166  0.9999960574352827
604.584 0.587432   1      2      90699673  2417569 406418  147.24759401351207 4115487.409606559  691855.3977311417  0.16811019664795504
```
I monitored the device specified by the rook's metadata with dstat along with this log. But my conclusion... is that there is no significant correlation between the compaction of rocksdb and a "rapid" decrease in speed. What has changed before and after the slowdown point (near offset 350) is that compaction occurs less frequently as the speed slows down, and that writes are reduced when observed with dstat. (I think this is natural)

I plan to analyze the other logs further. If my conclusion is wrong, please point it out. thank you!

---

I tried running it for a longer time. Contrary to my thought, frequency didn't seem to have a significant relationship.

|   start_offset |   compaction_time_seconds |   output_level |   num_output_files |   total_output_size |   num_input_records |   num_output_records |   output (MB/s) |      input (r/s) |   output (r/s) |   output/input ratio |
|----------------|---------------------------|----------------|--------------------|---------------------|---------------------|----------------------|-----------------|------------------|----------------|----------------------|
|        198.308 |                  0.577623 |              1 |                  2 |            91008535 |             1576263 |               406538 |       150.258   |      2.72888e+06 |       703812   |             0.257913 |
|        226.048 |                  0.919202 |              1 |                  2 |            90691697 |             2465910 |               406991 |        94.0929  |      2.68266e+06 |       442766   |             0.165047 |
|        239.001 |                  0.001722 |              1 |                  1 |                5697 |                 328 |                  148 |         3.1551  | 190476           |        85946.6 |             0.45122  |
|        241.209 |                  1.59318  |              1 |                  5 |           300523841 |              563467 |               562981 |       179.893   | 353675           |       353370   |             0.999137 |
|        242.537 |                  1.32175  |              1 |                  5 |           300852563 |              564047 |               563607 |       217.072   | 426742           |       426409   |             0.99922  |
|        243.885 |                  1.34387  |              1 |                  5 |           300409897 |              563216 |               562766 |       213.186   | 419101           |       418766   |             0.999201 |
|        254.529 |                  0.915761 |              1 |                  2 |            90563071 |             2446313 |               406519 |        94.3125  |      2.67134e+06 |       443914   |             0.166176 |
|        282.657 |                  0.994387 |              1 |                  2 |            90400754 |             2443754 |               406391 |        86.6995  |      2.45755e+06 |       408685   |             0.166298 |
|        310.473 |                  0.998574 |              1 |                  2 |            90708566 |             2434462 |               406458 |        86.63    |      2.43794e+06 |       407038   |             0.16696  |
|        315.973 |                  2.87988  |              1 |                  8 |           521696528 |              975226 |               975226 |       172.76    | 338635           |       338635   |             1        |
|        318.729 |                  2.74252  |              1 |                  8 |           521479348 |              974816 |               974816 |       181.338   | 355446           |       355446   |             1        |
|        321.361 |                  2.62247  |              1 |                  8 |           521833185 |              975492 |               975492 |       189.767   | 371975           |       371975   |             1        |
|        350.685 |                  0.430079 |              1 |                  2 |            90968130 |             2431164 |               407430 |       201.716   |      5.65283e+06 |       947338   |             0.167586 |
|        467.195 |                  0.528455 |              1 |                  2 |            90764313 |             2417242 |               406661 |       163.797   |      4.57417e+06 |       769528   |             0.168233 |
|        567.588 |                  0.001354 |              1 |                  1 |                5697 |                 157 |                  148 |         4.01262 | 115953           |       109306   |             0.942675 |
|        569.86  |                  1.83404  |              1 |                  8 |           537118924 |             1001696 |              1001696 |       279.294   | 546169           |       546169   |             1        |
|        571.496 |                  1.62678  |              1 |                  8 |           537173868 |             1001802 |              1001802 |       314.909   | 615818           |       615818   |             1        |
|        573.068 |                  1.56635  |              1 |                  8 |           537065722 |             1001606 |              1001606 |       326.994   | 639454           |       639454   |             1        |
|        573.508 |                  0.42611  |              2 |                  2 |           135710886 |              253654 |               253654 |       303.734   | 595278           |       595278   |             1        |
|        573.952 |                  0.441106 |              2 |                  2 |           135708683 |              253649 |               253648 |       293.403   | 575030           |       575027   |             0.999996 |
|        574.468 |                  0.513229 |              2 |                  2 |           135712315 |              253642 |               253641 |       252.179   | 494208           |       494206   |             0.999996 |
|        604.584 |                  0.587432 |              1 |                  2 |            90699673 |             2417569 |               406418 |       147.248   |      4.11549e+06 |       691855   |             0.16811  |
|        754.109 |                  0.506633 |              1 |                  2 |            90757337 |             2414447 |               406637 |       170.84    |      4.76567e+06 |       802626   |             0.168418 |
|        896.741 |                  0.455104 |              1 |                  2 |            90720870 |             2414444 |               406503 |       190.106   |      5.30526e+06 |       893209   |             0.168363 |
|        949.073 |                  1.8624   |              1 |                  8 |           490062668 |              912400 |               912400 |       250.945   | 489905           |       489905   |             1        |
|        950.661 |                  1.58112  |              1 |                  8 |           489253569 |              910908 |               910908 |       295.1     | 576115           |       576115   |             1        |
|        952.241 |                  1.57359  |              1 |                  8 |           490057778 |              912393 |               912393 |       296.999   | 579816           |       579816   |             1        |
|        953.001 |                  0.753394 |              2 |                  3 |           203537297 |              380028 |               380028 |       257.645   | 504421           |       504421   |             1        |
|        953.753 |                  0.749621 |              2 |                  3 |           203536846 |              380027 |               380027 |       258.941   | 506959           |       506959   |             1        |
|        954.497 |                  0.738912 |              2 |                  3 |           203541867 |              380019 |               380019 |       262.701   | 514295           |       514295   |             1        |
|        955.161 |                  0.659853 |              2 |                  3 |           203517508 |              380026 |               380026 |       294.14    | 575925           |       575925   |             1        |
|        955.749 |                  0.584168 |              2 |                  3 |           203514020 |              380032 |               380032 |       332.244   | 650553           |       650553   |             1        |
|        956.345 |                  0.593092 |              2 |                  3 |           203519475 |              380035 |               380035 |       327.253   | 640769           |       640769   |             1        |
|        957.037 |                  0.686311 |              2 |                  3 |           203476738 |              379950 |               379950 |       282.744   | 553612           |       553612   |             1        |
|        957.701 |                  0.660492 |              2 |                  3 |           203470968 |              379943 |               379943 |       293.789   | 575242           |       575242   |             1        |
|        958.373 |                  0.669117 |              2 |                  3 |           203473048 |              379946 |               379946 |       290.005   | 567832           |       567832   |             1        |
|        959.069 |                  0.691467 |              2 |                  3 |           203436867 |              379864 |               379864 |       280.581   | 549360           |       549360   |             1        |
|        959.657 |                  0.584352 |              2 |                  3 |           203427117 |              379851 |               379851 |       331.997   | 650038           |       650038   |             1        |
|        960.309 |                  0.649861 |              2 |                  3 |           203433946 |              379865 |               379865 |       298.54    | 584533           |       584533   |             1        |
|       1037.89  |                  0.678762 |              1 |                  2 |            90863747 |             2413422 |               407020 |       127.665   |      3.55562e+06 |       599651   |             0.168648 |
|       1164.19  |                  0.556377 |              1 |                  2 |            89989302 |             2438727 |               401141 |       154.249   |      4.38323e+06 |       720988   |             0.164488 |
|       1272.27  |                  0.00183  |              1 |                  1 |                5641 |                 159 |                  146 |         2.93971 |  86885.2         |        79781.4 |             0.918239 |
|       1274.28  |                  1.64445  |              1 |                  8 |           499922044 |              974756 |               958921 |       289.922   | 592754           |       583125   |             0.983755 |
|       1275.76  |                  1.46466  |              1 |                  8 |           499222594 |              973882 |               957845 |       325.055   | 664920           |       653971   |             0.983533 |
|       1277.4   |                  1.63778  |              1 |                  8 |           499709137 |              974694 |               958672 |       290.979   | 595132           |       585349   |             0.983562 |
|       1278.08  |                  0.67253  |              2 |                  3 |           197147073 |              394232 |               368136 |       279.562   | 586192           |       547390   |             0.933805 |
|       1278.77  |                  0.686701 |              2 |                  3 |           197106782 |              394276 |               368063 |       273.737   | 574160           |       535987   |             0.933516 |
|       1279.23  |                  0.457522 |              1 |                  2 |            89732178 |             2439925 |               400197 |       187.041   |      5.33291e+06 |       874705   |             0.16402  |
|       1279.93  |                  0.692498 |              2 |                  3 |           197102257 |              394337 |               368040 |       271.44    | 569441           |       531467   |             0.933313 |
|       1280.46  |                  0.532112 |              2 |                  3 |           191000728 |              381823 |               356399 |       342.32    | 717561           |       669782   |             0.933414 |
|       1281.05  |                  0.582235 |              2 |                  3 |           190913621 |              381899 |               356238 |       312.708   | 655919           |       611846   |             0.932807 |
|       1281.62  |                  0.564145 |              2 |                  3 |           190971670 |              381757 |               356344 |       322.833   | 676700           |       631653   |             0.933431 |
|       1282.25  |                  0.624516 |              2 |                  3 |           190943760 |              356228 |               356228 |       291.583   | 570407           |       570407   |             1        |
|       1282.84  |                  0.583463 |              2 |                  3 |           190851368 |              356066 |               356066 |       311.948   | 610263           |       610263   |             1        |
|       1283.4   |                  0.562001 |              2 |                  3 |           190909475 |              356169 |               356169 |       323.959   | 633752           |       633752   |             1        |
|       1284.02  |                  0.620917 |              2 |                  3 |           203564135 |              379798 |               379798 |       312.657   | 611673           |       611673   |             1        |
|       1284.66  |                  0.632677 |              2 |                  3 |           203571437 |              379812 |               379812 |       306.856   | 600325           |       600325   |             1        |
|       1285.26  |                  0.60105  |              2 |                  3 |           203566491 |              379807 |               379807 |       322.995   | 631906           |       631906   |             1        |
|       1430.2   |                  0.461121 |              1 |                  2 |            89283913 |             2440574 |               397304 |       184.654   |      5.2927e+06  |       861605   |             0.162791 |
|       1554.9   |                  0.538048 |              1 |                  2 |            89328782 |             2442588 |               397476 |       158.333   |      4.53972e+06 |       738737   |             0.162727 |
|       1629     |                  1.8959   |              1 |                  8 |           516680641 |              986645 |               978640 |       259.901   | 520410           |       516188   |             0.991887 |
|       1630.6   |                  1.59285  |              1 |                  8 |           516084064 |              985854 |               977757 |       308.992   | 618926           |       613843   |             0.991787 |
|       1632.19  |                  1.5775   |              1 |                  8 |           516590635 |              986642 |               978592 |       312.303   | 625446           |       620343   |             0.991841 |
|       1632.52  |                  0.329432 |              2 |                  2 |           103630563 |              193332 |               193332 |       300.001   | 586865           |       586865   |             1        |
|       1632.88  |                  0.361199 |              2 |                  2 |           103088965 |              192327 |               192327 |       272.186   | 532468           |       532468   |             1        |
|       1633.22  |                  0.33615  |              2 |                  2 |           103491120 |              193073 |               193073 |       293.609   | 574366           |       574366   |             1        |
|       1633.8   |                  0.577307 |              2 |                  3 |           195031957 |              396644 |               363800 |       322.18    | 687059           |       630167   |             0.917195 |
|       1634.36  |                  0.554075 |              2 |                  3 |           194890750 |              396883 |               363541 |       335.446   | 716298           |       656122   |             0.91599  |
|       1634.95  |                  0.582429 |              2 |                  3 |           194982072 |              396760 |               363710 |       319.265   | 681216           |       624471   |             0.9167   |
|       1635.6   |                  0.644126 |              2 |                  3 |           203530137 |              379682 |               379682 |       301.341   | 589453           |       589453   |             1        |
|       1636.29  |                  0.690095 |              2 |                  3 |           203539016 |              379685 |               379684 |       281.28    | 550192           |       550191   |             0.999997 |
|       1637.01  |                  0.719006 |              2 |                  3 |           203534108 |              379679 |               379678 |       269.963   | 528061           |       528060   |             0.999997 |
|       1637.94  |                  0.927455 |              2 |                  4 |           271318702 |              506065 |               506065 |       278.989   | 545649           |       545649   |             1        |
|       1638.78  |                  0.832911 |              2 |                  4 |           271332632 |              506085 |               506085 |       310.673   | 607610           |       607610   |             1        |
|       1639.75  |                  0.959941 |              2 |                  4 |           271325454 |              506081 |               506081 |       269.554   | 527200           |       527200   |             1        |
|       1640.66  |                  0.905929 |              2 |                  4 |           271235112 |              505911 |               505911 |       285.53    | 558444           |       558444   |             1        |
|       1641.58  |                  0.91619  |              2 |                  4 |           271256839 |              505948 |               505948 |       282.355   | 552230           |       552230   |             1        |
|       1642.51  |                  0.92607  |              2 |                  4 |           271240687 |              505924 |               505924 |       279.326   | 546313           |       546313   |             1        |
|       1699.71  |                  0.737009 |              1 |                  2 |            89446152 |             2444696 |               397917 |       115.741   |      3.31705e+06 |       539908   |             0.162767 |
|       1835.15  |                  0.496634 |              1 |                  2 |            89329543 |             2445644 |               397467 |       171.537   |      4.92444e+06 |       800322   |             0.16252  |
|       1956.37  |                  0.57818  |              1 |                  2 |            89257163 |             2445812 |               396945 |       147.225   |      4.23019e+06 |       686542   |             0.162296 |
|       1982.22  |                  0.002782 |              1 |                  1 |                5609 |                 158 |                  145 |         1.92277 |  56793.7         |        52120.8 |             0.917722 |
|       1984.26  |                  1.65109  |              1 |                  8 |           495759415 |              922827 |               922827 |       286.352   | 558920           |       558920   |             1        |
|       1985.83  |                  1.56203  |              1 |                  8 |           496916665 |              924989 |               924988 |       303.385   | 592171           |       592170   |             0.999999 |
|       1987.52  |                  1.68087  |              1 |                  8 |           495120062 |              921641 |               921640 |       280.917   | 548313           |       548312   |             0.999999 |
|       1988.23  |                  0.710482 |              2 |                  3 |           203538855 |              379654 |               379654 |       273.209   | 534361           |       534361   |             1        |
|       1988.91  |                  0.667849 |              2 |                  3 |           203536093 |              379646 |               379646 |       290.645   | 568461           |       568461   |             1        |
|       1989.57  |                  0.654229 |              2 |                  3 |           203541159 |              379660 |               379660 |       296.703   | 580317           |       580317   |             1        |
|       1990.43  |                  0.856645 |              2 |                  4 |           258738759 |              482551 |               482551 |       288.045   | 563303           |       563303   |             1        |
|       1991.33  |                  0.902283 |              2 |                  4 |           258648832 |              482387 |               482387 |       273.381   | 534629           |       534629   |             1        |
|       1992.18  |                  0.845867 |              2 |                  4 |           258710130 |              482504 |               482504 |       291.683   | 570425           |       570425   |             1        |
|       1993.07  |                  0.877668 |              2 |                  4 |           271385802 |              506159 |               506159 |       294.888   | 576709           |       576709   |             1        |
|       1994.01  |                  0.938822 |              2 |                  4 |           271394509 |              506172 |               506172 |       275.688   | 539157           |       539157   |             1        |
|       1994.87  |                  0.858627 |              2 |                  4 |           271388612 |              506162 |               506162 |       301.43    | 589502           |       589502   |             1        |
|       1996.01  |                  1.13594  |              2 |                  5 |           339143978 |              632508 |               632508 |       284.728   | 556817           |       556817   |             1        |
|       1997.12  |                  1.09545  |              2 |                  5 |           339151786 |              632528 |               632528 |       295.259   | 577416           |       577416   |             1        |
|       1998.35  |                  1.22582  |              2 |                  5 |           339157000 |              632523 |               632523 |       263.86    | 515998           |       515998   |             1        |
|       2106.04  |                  0.590092 |              1 |                  2 |            89492948 |             2452626 |               397851 |       144.634   |      4.15635e+06 |       674219   |             0.162214 |
|       2248.42  |                  0.469359 |              1 |                  2 |            89326262 |             2452627 |               397223 |       181.499   |      5.22548e+06 |       846310   |             0.161958 |
|       2349.06  |                  1.85958  |              1 |                  8 |           517930846 |              964113 |               964113 |       265.618   | 518458           |       518458   |             1        |
|       2350.84  |                  1.77426  |              1 |                  8 |           518353906 |              964900 |               964900 |       278.618   | 543833           |       543833   |             1        |
|       2352.77  |                  1.92541  |              1 |                  8 |           517032554 |              962432 |               962432 |       256.092   | 499859           |       499859   |             1        |
|       2353.69  |                  0.90644  |              2 |                  4 |           271291649 |              505972 |               505972 |       285.429   | 558197           |       558197   |             1        |
|       2354.7   |                  1.003    |              2 |                  4 |           271305296 |              506005 |               506005 |       257.963   | 504491           |       504491   |             1        |
|       2355.86  |                  1.14816  |              2 |                  4 |           271295639 |              505977 |               505977 |       225.341   | 440684           |       440684   |             1        |
|       2356.84  |                  0.979729 |              2 |                  4 |           271379708 |              506168 |               506168 |       264.163   | 516641           |       516641   |             1        |
|       2357.94  |                  1.09042  |              2 |                  4 |           271377723 |              506141 |               506140 |       237.345   | 464170           |       464169   |             0.999998 |
|       2358.95  |                  1.00017  |              2 |                  4 |           271384297 |              506146 |               506145 |       258.767   | 506058           |       506057   |             0.999998 |
|       2359.7   |                  0.749341 |              2 |                  3 |           195878809 |              365296 |               365296 |       249.292   | 487490           |       487490   |             1        |
|       2360.45  |                  0.748947 |              2 |                  3 |           196096818 |              365702 |               365702 |       249.701   | 488288           |       488288   |             1        |
|       2361.13  |                  0.673154 |              2 |                  3 |           194905822 |              363481 |               363481 |       276.128   | 539967           |       539967   |             1        |
|       2362.34  |                  1.20132  |              2 |                  5 |           339003391 |              632234 |               632234 |       269.119   | 526282           |       526282   |             1        |
|       2363.58  |                  1.22817  |              2 |                  5 |           339026001 |              632276 |               632276 |       263.253   | 514810           |       514810   |             1        |
|       2364.89  |                  1.30171  |              2 |                  5 |           339008197 |              632244 |               632244 |       248.368   | 485702           |       485702   |             1        |
|       2366.07  |                  1.17547  |              2 |                  5 |           338919778 |              632057 |               632057 |       274.97    | 537705           |       537705   |             1        |
|       2378.94  |                  0.646041 |              1 |                  2 |            89487876 |             2453649 |               397830 |       132.1     |      3.79798e+06 |       615797   |             0.162138 |
|       2512.16  |                  0.524455 |              1 |                  2 |            89331427 |             2453943 |               397120 |       162.441   |      4.67903e+06 |       757205   |             0.161829 |
|       2650.18  |                  0.496938 |              1 |                  2 |            89360125 |             2452986 |               397245 |       171.491   |      4.9362e+06  |       799385   |             0.161943 |
|       2705.86  |                  0.002095 |              1 |                  1 |                5609 |                 154 |                  145 |         2.5533  |  73508.4         |        69212.4 |             0.941558 |
|       2708     |                  1.78988  |              1 |                  8 |           493945964 |              919459 |               919459 |       263.182   | 513700           |       513700   |             1        |
|       2709.86  |                  1.85205  |              1 |                  9 |           561359087 |             1044995 |              1044995 |       289.06    | 564237           |       564237   |             1        |
|       2711.75  |                  1.87441  |              1 |                  9 |           561961720 |             1046118 |              1046118 |       285.918   | 558105           |       558105   |             1        |
|       2712.74  |                  0.986309 |              2 |                  4 |           271387595 |              506114 |               506114 |       262.408   | 513139           |       513139   |             1        |
|       2713.66  |                  0.906437 |              2 |                  4 |           271382253 |              506113 |               506113 |       285.525   | 558354           |       558354   |             1        |
|       2714.57  |                  0.905109 |              2 |                  4 |           271387811 |              506122 |               506122 |       285.95    | 559183           |       559183   |             1        |
|       2715.63  |                  1.05477  |              2 |                  5 |           339040463 |              632263 |               632263 |       306.544   | 599430           |       599430   |             1        |
|       2716.56  |                  0.922261 |              2 |                  4 |           271388183 |              506126 |               506126 |       280.632   | 548788           |       548788   |             1        |
|       2717.59  |                  1.02873  |              2 |                  4 |           271387986 |              506135 |               506135 |       251.588   | 492001           |       492001   |             1        |
|       2718.89  |                  1.2883   |              2 |                  5 |           339054806 |              632310 |               632310 |       250.988   | 490810           |       490810   |             1        |
|       2721.15  |                  2.25127  |              2 |                  5 |           339031196 |              632258 |               632258 |       143.619   | 280845           |       280845   |             1        |
|       2724.66  |                  3.49747  |              2 |                  5 |           339045183 |              632260 |               632260 |        92.4494  | 180777           |       180777   |             1        |
|       2728.92  |                  4.23035  |              2 |                  6 |           406669639 |              758382 |               758382 |        91.6781  | 179272           |       179272   |             1        |
|       2732.75  |                  3.77327  |              2 |                  5 |           338745882 |              631707 |               631707 |        85.6162  | 167416           |       167416   |             1        |
|       2736.91  |                  4.13086  |              2 |                  5 |           339245596 |              632649 |               632649 |        78.3202  | 153152           |       153152   |             1        |
|       2742.18  |                  5.24851  |              2 |                  6 |           406995588 |              759056 |               759055 |        73.9526  | 144623           |       144623   |             0.999999 |
|       2744.53  |                  2.31148  |              2 |                  5 |           337736073 |              629840 |               629840 |       139.344   | 272484           |       272484   |             1        |
|       2845.1   |                  1.05653  |              1 |                  2 |            89319552 |             2454000 |               397091 |        80.6242  |      2.3227e+06  |       375845   |             0.161814 |

And compaction takes place in succession. But since this is already very slow. There doesn't seem to be much difference when the cluster slows down. No warning was printed. I'll have to find another reason...
